server/jobs/job_receiver_registry.go

```diff
@@ -0,0 +1,58 @@
+package jobs
+
+import "sync"
+
+type ReceiverRegistry interface {
+	AddReceiver(jobID string, ch chan string)
+	RemoveReceiver(jobID string, ch chan string)
+	GetReceivers(jobID string) map[chan string]bool
+	CloseAndRemoveReceiversForJob(jobID string)
+}
+
+type receiverRegistry struct {
+	receivers map[string]map[chan string]bool
+	lock      sync.RWMutex
+}
+
+func NewReceiverRegistry() ReceiverRegistry {
```

Comment:
Why does this need to return the interface? Let's return the implementation as it's better Go practice.

Reply:
Done!